plugins/osdn/ovs/plugin.go

```diff
@@ -124,25 +125,77 @@ func (plugin *ovsPlugin) getVNID(namespace string) (string, error) {
 	return "0", nil
 }
 
+var minRsrc = resource.MustParse("1k")
+var maxRsrc = resource.MustParse("1P")
+
+func validateBandwidthIsReasonable(rsrc *resource.Quantity) error {
+	if rsrc.Value() < minRsrc.Value() {
+		return fmt.Errorf("resource is unreasonably small (< 1kbit)")
+	}
+	if rsrc.Value() > maxRsrc.Value() {
+		return fmt.Errorf("resource is unreasonably large (> 1Pbit)")
+	}
+	return nil
+}
+
+func extractBandwidthResources(pod *api.Pod) (ingress, egress string, err error) {
+	str, found := pod.Annotations["kubernetes.io/ingress-bandwidth"]
+	if found {
+		ingressQuantity, err := resource.ParseQuantity(str)
+		if err != nil {
+			return "", "", err
+		}
+		if err := validateBandwidthIsReasonable(ingressQuantity); err != nil {
+			return "", "", err
+		}
+		ingress = fmt.Sprintf("%d", ingressQuantity.Value())
```

Comment:
it seems wrong to have this function return numbers as strings, just because we happen to want strings later on